### PR TITLE
Add mixer v1 filter config to v2 listeners

### PR DESCRIPTION
**Pull request by ostromart**
Some parts still missing but this a good starting point for testing.

### CODE REVIEW

**Comment by narehayrapetyan** [comment]
I wasn't able to push to vendor to add a cloudwatch dependency. As far as I understand it I need to push to a new branch in vendor and not my fork correct? When I try to do that I get permission denied.

File: pilot/pkg/networking/v1alpha3/listener.go [hunk 1]
```diff
@@ -236,23 +237,13 @@ func buildSidecarInboundListeners(env model.Environment, node model.Proxy,
...
-			//	})
-			//}
-
 		default:
 			log.Debugf("Unsupported inbound protocol %v for port %#v", protocol, instance.Endpoint.ServicePort)
 		}
 
+		listenerOpts.networkFilters = append(listenerOpts.networkFilters,
+			buildMixerFilter(env, node, proxyInstances, protocol, false))
+
```
Comment (rshriram) [inline pilot/pkg/networking/v1alpha3/listener.go]: this is for TCP only. For HTTP, we need to add a HTTP filter.

File: pilot/pkg/networking/v1alpha3/listener.go [hunk 2]
```diff
@@ -315,6 +307,8 @@ func buildSidecarOutboundListeners(env model.Environment, node model.Proxy,
 				}
 				listenerOpts.ip = listenAddress
 				listenerOpts.networkFilters = buildOutboundNetworkFilters(clusterName, addresses, servicePort)
+				listenerOpts.networkFilters = append(listenerOpts.networkFilters,
+					buildMixerFilter(env, node, proxyInstances, protocol, true))
```
Comment (rshriram) [inline pilot/pkg/networking/v1alpha3/listener.go]: no mixer in outbound path.

File: pilot/pkg/networking/v1alpha3/listener.go [hunk 3]
```diff
@@ -336,6 +330,8 @@ func buildSidecarOutboundListeners(env model.Environment, node model.Proxy,
 					direction:        operation,
 					authnPolicy:      nil, /* authn policy is not needed for outbound listener */
 				}
+				listenerOpts.networkFilters = append(listenerOpts.networkFilters,
+					buildMixerFilter(env, node, proxyInstances, protocol, true))
```
Comment (rshriram) [inline pilot/pkg/networking/v1alpha3/listener.go]: same as above. Network filters are not parsed for building HTTP listeners..

File: pilot/pkg/networking/v1alpha3/listener.go [hunk 4]
```diff
@@ -396,6 +392,33 @@ func buildMgmtPortListeners(managementPorts model.PortList, managementIP string)
...
+	mesh := env.Mesh
+	config := env.IstioConfigStore
+	if mesh.MixerCheckServer == "" && mesh.MixerReportServer == "" {
+		return listener.Filter{}
+	}
+
+	mixerFilter := listener.Filter{}
+	switch protocol {
+	case model.ProtocolHTTP, model.ProtocolHTTP2, model.ProtocolGRPC:
+		config := v1.BuildHTTPMixerFilterConfig(mesh, node, proxyInstances, outbound, config)
```
Comment (rshriram) [inline pilot/pkg/networking/v1alpha3/listener.go]: Lets please not cross into v1. I would much rather copy that code here, the way I did for TCP proxy routes. I would like to be able to delete all proxy/v1 and still see the mesh functioning.

**Comment by rshriram** [comment]
merging to continue further work

**Pull request merged**